functions/go/set-image/image_transformer_test.go

```diff
@@ -30,15 +30,16 @@ func runImageTransformerResults(input, config string) (*fn.ResourceList, error)
 	}
 	rl.FunctionConfig = functionConfig
 	in, _ := rl.ToYAML()
-	out, err := fn.Run(fn.ResourceListProcessorFunc(setImageTags), in)
+	out, err := fn.Run(fn.ResourceListProcessorFunc(SetImageTagSDK), in)
```

Comment:
Suggest using the `fn.Runner` rather than `ResourceListProcessorFunc`. Here's an [example ](https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/master/go/fn/examples/example_asmain_runner_test.go)

Reply:
I am checking out `Runner` interface, it seems the `*fn.Context` does not provide good result logging. Its related functions only provide message and severity, no further information about `fn.ResourceRef`, `fn.Field` or `fn.File`

Reply:
If so, we shall add a "ConfigObjectResult" method to `Context`

Reply:
Oh I see, I missed this important method.

Reply:
`ConfigObjectResult` does not contain `field`, and would miss some of the information. Should I add it in the function? Or it is ok to leave out those information `path`, `currentValue`, `proposedValue`.

Reply:
There are some other issues with this `fn.Runner`, I am not sure how to test it in unit tests. Because I cannot create `Context`, its field is not exported, there is no way for me to test its Results because I cannot create `Context`. In e2e test, this creation of `Context` is completed by `AsMain`.

Reply:
There are more issues with `fn.Runner` than I think. Its config function is not very generalized, it can only deal with map type, such as `SetLabels` kind, but for `SetImage`, the type is Image instead of map. More debugging error details in 1on1 doc.